### PR TITLE
[WIP] Coh 91 create event when question round is issued

### PR DESCRIPTION
**Pull request by phillipDenness**

### CODE REVIEW

**Comment by phillipDenness** [comment]
Will address this comment from previous PR - ```Thanks to yet another HLD change, the state of the questions in the round need to be set to "questions_issue_pending" instead of "questions_issued". Only after the notification job actually sends the message to the endpoint does the questions and online hearing get updated```

File: src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/ApiSteps.java [hunk 1]
```diff
@@ -106,7 +110,15 @@ public void cleanUp() {
...
-
+        if(testContext.getScenarioContext().getSessionEventForwardingRegisters() != null) {
+            for (SessionEventForwardingRegister sessionEventForwardingRegister : testContext.getScenarioContext().getSessionEventForwardingRegisters()) {
+                try {
+                    sessionEventForwardingRegisterRepository.delete(sessionEventForwardingRegister);
+                } catch (DataIntegrityViolationException e) {
+                    log.error("Failure may be due to foreign key. This is okay because the online hearing will be deleted elsewhere.");
+                }
+            }
+        }
```
Comment (johnhosolirius) [inline src/aat/java/uk/gov/hmcts/reform/coh/functional/bdd/steps/ApiSteps.java]: As a side note. All this duplication is unnecessary. It's fine to have this clean up code in just one of the steps classes and it get run for all. I keep meaning to refactor this.

File: src/main/java/uk/gov/hmcts/reform/coh/controller/AnswerController.java [hunk 2]
```diff
@@ -87,9 +86,11 @@ public ResponseEntity createAnswer(UriComponentsBuilder uriBuilder, @PathVariabl
         try {
 
             Optional<Question> optionalQuestion = questionService.retrieveQuestionById(questionId);
+
             // If a question exists, then it must be in the issued state to be answered
             if (!optionalQuestion.isPresent()
-                    || !optionalQuestion.get().getQuestionState().getState().equals(QuestionStates.ISSUED.getStateName())) {
+                    || !optionalQuestion.get().getQuestionState().getState().equals(QuestionStates.ISSUED_PENDING.getStateName())
```
Comment (johnhosolirius) [inline src/main/java/uk/gov/hmcts/reform/coh/controller/AnswerController.java]: If it's issue pending, then the question hasn't been sent

Comment (phillipDenness) [reply src/main/java/uk/gov/hmcts/reform/coh/controller/AnswerController.java]: When I asked you if an appellant can answer a question in issued pending state you said they can :(

Comment (johnhosolirius) [reply src/main/java/uk/gov/hmcts/reform/coh/controller/AnswerController.java]: I did? I don't remember. Apologies if I did.

Comment (phillipDenness) [reply src/main/java/uk/gov/hmcts/reform/coh/controller/AnswerController.java]: For the answer cucumber tests to work the question must be question_issued. For this to happen the question issued event must be processing by the scheduler and update the question_issued_pending to question_issued after forwarding the notification. At the moment this functionality does not exist.

Comment (phillipDenness) [reply src/main/java/uk/gov/hmcts/reform/coh/controller/AnswerController.java]: So it's up to you whether to wait until someone has developed the functionality for the question_issued_pending state to be updated question_issued state. Or to merge it now and i'll add fix the state later

**Pull request closed without merging**